ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/completion/EmptyLedgerIT.scala

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.sandbox.services.completion
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+import com.daml.ledger.api.testing.utils.SuiteResourceManagementAroundAll
+import com.daml.ledger.api.v1.command_completion_service.{
+  CommandCompletionServiceGrpc,
+  CompletionEndRequest,
+  CompletionStreamRequest,
+  CompletionStreamResponse
+}
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.platform.sandbox.SandboxBackend
+import com.daml.platform.sandbox.config.SandboxConfig
+import com.daml.platform.sandbox.services.TestCommands
+import com.daml.platform.sandboxnext.SandboxNextFixture
+import com.daml.platform.testing.StreamConsumer
+import org.scalatest.{AsyncWordSpec, Inspectors, Matchers}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scalaz.syntax.tag._
+
+class EmptyLedgerIT
+    extends AsyncWordSpec
+    with Matchers
+    with Inspectors
+    with SandboxNextFixture
+    with SandboxBackend.Postgresql
+    with TestCommands
+    with SuiteResourceManagementAroundAll {
+
+  // Start with empty daml packages and a large configuration delay, such that we can test the API's behavior
+  // on an empty index
+  override protected def config: SandboxConfig =
+    super.config.copy(
+      damlPackages = List.empty,
+      ledgerConfig = super.config.ledgerConfig.copy(
+        initialConfigurationSubmitDelay = Duration.ofDays(5)
+      ),
+      implicitPartyAllocation = false
+    )
+
+  private[this] val applicationId = "CompletionServiceIT"
+
+  // How long it takes to download the entire completion stream.
+  // Because the stream does not terminate, we use a timeout to determine when the stream
+  // is done emitting elements.
+  private[this] val completionTimeout = FiniteDuration(2, TimeUnit.SECONDS)
+
+  private[this] def completionsFromOffset(
+      completionService: CommandCompletionServiceGrpc.CommandCompletionServiceStub,
+      ledgerId: String,
+      parties: Seq[String],
+      offset: LedgerOffset,
+  ): Future[Vector[String]] = {
+    new StreamConsumer[CompletionStreamResponse](
+      completionService.completionStream(
+        CompletionStreamRequest(ledgerId, applicationId, parties, Some(offset)),
+        _
+      )
+    ).within(completionTimeout)
+      .map(_.flatMap(_.completions).map(_.commandId))
+  }
+
+  "CommandCompletionService gives sensible ledger end on an empty ledger" in {
+
+    val partyA = "partyA"
+    val lid = ledgerId().unwrap
+    val completionService = CommandCompletionServiceGrpc.stub(channel)
+    for {
+      end <- completionService.completionEnd(CompletionEndRequest(lid))
+      completions <- completionsFromOffset(completionService, lid, List(partyA), end.getOffset)
+    } yield {
+      completions shouldBe empty
```

Comment:
What happens when a transaction is written to the ledger after opening the stream? Does using `end.getOffset` get that completion?

Reply:
`.getOffset` is just an unsafe value getter for the (optional) protobuf field, so it should return the empty ledger begin regardless of what happens later. The cases where there are existing transactions before the end is requested are covered by the `CompletionServiceIT`

Reply:
I'm specifically wondering what happens in the case where we return `Offset.begin` rather than a value from the database, which we don't test elsewhere.

Reply:
Discussed with @oggy-; this flags further issues so we're going to open a separate issue.

Reply:
Tracked now in #6640